back/node_watcher/src/tasks/createTreasury.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018-2020 @paritytech/nomidot authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { ApiPromise } from '@polkadot/api';
+import { BlockNumber, TreasuryProposal, Hash } from '@polkadot/types/interfaces';
+import { logger } from '@polkadot/util';
+
+import { treasuryProposalStatus } from '../util/statuses';
+import { prisma } from '../generated/prisma-client';
+import { filterEvents } from '../util/filterEvents';
+import { Cached, NomidotTreasury, NomidotTreasuryEvent, NomidotTreasuryRawEvent, Task } from './types';
+
+const l = logger('Task: Treasury');
+
+/*
+ *  ======= Table (Treasury) ======
+ */
+const createTreasury: Task<NomidotTreasury[]> = {
+  name: 'createTreasury',
+  read: async (
+    blockHash: Hash,
+    cached: Cached,
+    api: ApiPromise
+  ): Promise<NomidotTreasury[]> => {
+    const { events } = cached;
+
+    const treasuryEvents = filterEvents(events, 'treasury', treasuryProposalStatus.PROPOSED);
+
+    const results: NomidotTreasury[] = [];
+
+    await Promise.all(
+      treasuryEvents.map(async ({ event: { data, typeDef } }) => {
+        const treasuryRawEvent: NomidotTreasuryRawEvent = data.reduce(
+          (prev, curr, index) => {
+            const type = typeDef[index].type;
+
+            return {
+              ...prev,
+              [type]: curr.toJSON(),
+            };
+          },
+          {}
+        );
+
+        if (
+          !treasuryRawEvent.ProposalIndex &&
+          treasuryRawEvent.ProposalIndex !== 0
+        ) {
+          l.error(
+            `Expected ProposalIndex missing in the event: ${treasuryRawEvent.ProposalIndex}`
+          );
+          return null;
+        }
+
+        const treasuryProposalRaw = await api.query.treasury.proposals(treasuryRawEvent.ProposalIndex);
+        const treasuryProposal: TreasuryProposal = api.createType('TreasuryProposal', treasuryProposalRaw.toU8a(true));
+
+        const proposalArguments: NomidotTreasuryEvent = {
+          proposalId: treasuryRawEvent.ProposalIndex,
+        };
```

Comment:
Not sure what those lines help for. I'd use the `treasuryRawEvent.ProposalIndex` right away.
I might have done something similar in another task to make things type safe, but that might be just over-complicating things.

Reply:
thanks, removed. It was unnecessary